src/semidiscretization/semidiscretization_hyperbolic_parabolic.jl

```diff
@@ -0,0 +1,256 @@
+
+  equations::Equations
+  equations_parabolic::EquationsParabolic
+
+  # This guy is a bit messy since we abuse it as some kind of "exact solution"
+  # although this doesn't really exist...
+  initial_condition::InitialCondition
+
+  boundary_conditions::BoundaryConditions
+  parabolic_boundary_conditions::BoundaryConditionsParabolic
+
+  source_terms::SourceTerms
+
+  solver::Solver
+  # TODO: introduce `solver_parabolic` for future specialization.
+
+  cache::Cache
+  parabolic_cache::CacheParabolic
+
+  performance_counter::PerformanceCounter
+
+  function SemidiscretizationHyperbolicParabolic{Mesh, Equations, EquationsParabolic, InitialCondition, BoundaryConditions, BoundaryConditionsParabolic, SourceTerms, Solver, Cache, CacheParabolic}(
+      mesh::Mesh, equations::Equations, equations_parabolic::EquationsParabolic, initial_condition::InitialCondition,
+      boundary_conditions::BoundaryConditions, parabolic_boundary_conditions::BoundaryConditionsParabolic,
+      source_terms::SourceTerms, solver::Solver, cache::Cache, parabolic_cache::CacheParabolic) where {Mesh, Equations, EquationsParabolic, InitialCondition, BoundaryConditions, BoundaryConditionsParabolic, SourceTerms, Solver, Cache, CacheParabolic}
+    @assert ndims(mesh) == ndims(equations)
+
+    # Todo: assert nvariables(equations)==nvariables(equations_parabolic)
+
+    performance_counter = PerformanceCounter()
+
+    new(mesh, equations, equations_parabolic, initial_condition,
+        boundary_conditions, parabolic_boundary_conditions,
+        source_terms, solver, cache, parabolic_cache, performance_counter)
+  end
+end
+
+"""
+    SemidiscretizationHyperbolicParabolic(mesh, both_equations, initial_condition, solver;
+                                          source_terms=nothing,
+                                          both_boundary_conditions=(boundary_condition_periodic, boundary_condition_periodic),
+                                          RealT=real(solver),
+                                          uEltype=RealT,
+                                          both_initial_caches=(NamedTuple(), NamedTuple()))
+
+Construct a semidiscretization of a hyperbolic-parabolic PDE.
+"""
+function SemidiscretizationHyperbolicParabolic(mesh, equations::Tuple,
+                                               initial_condition, solver;
+                                               source_terms=nothing,
+                                               boundary_conditions=(boundary_condition_periodic, boundary_condition_periodic),
+                                               # `RealT` is used as real type for node locations etc.
+                                               # while `uEltype` is used as element type of solutions etc.
+                                               RealT=real(solver), uEltype=RealT,
+                                               initial_caches=(NamedTuple(), NamedTuple()))
+
+  hyperbolic_equations, equations_parabolic = equations
```

Comment:
This change I'm not sure about, because this ruins the `mesh_equations_dg_cache(semi)` dispatch structure. `SemidiscretizationHyperbolicParabolic` works because `@unpack mesh, equations, solver, cache = semi` doesn't change. 

We could change the name and then overload `getproperties` for `:equations`, however - how does that sound?

Reply:
I don't see how this changes with flipping the name, since `equations_hyperbolic` is also new to this `Semidiscretization`, isn't it? Here, we are just unpacking a tuple into two variables, which we can name anything we want. Or am I missing something here?

Reply:
Oops, I was looking at the wrong file. Yes, this one's fine - thanks for the catch!